### PR TITLE
Wrapping a timer on device flow auth

### PR DESCRIPTION
**Pull request by shubham90**

### CODE REVIEW

File: Build.props [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <CredentialProviderVersion>0.1.8</CredentialProviderVersion>
```
Comment (shubham90) [inline Build.props]: Is this the only place we bump the version?

File: CredentialProvider.Microsoft/Resources.resx [hunk 2]
```diff
@@ -383,4 +387,10 @@ Windows Integrated Authentication
   <data name="NotRunningBearerTokenProvider" xml:space="preserve">
     <value>Not running bearer token provider '{0}'</value>
   </data>
+  <data name="DeviceFlowTimedOut" xml:space="preserve">
+    <value>Device flow authentication failed. User was presented with device flow, but didn't react within {0} seconds</value>
```
Comment (zarenner) [inline CredentialProvider.Microsoft/Resources.resx]: nit: Period after first sentence but not second.

File: CredentialProvider.Microsoft/Util/EnvUtil.cs [hunk 3]
```diff
@@ -21,6 +21,7 @@ public static class EnvUtil
         public const string AdalFileCacheEnvVar = "NUGET_CREDENTIALPROVIDER_ADAL_FILECACHE_ENABLED";
         public const string PpeHostsEnvVar = "NUGET_CREDENTIALPROVIDER_ADAL_PPEHOSTS";
 
+        public const string DeviceFlowTimeoutEnvVar = "NUGET_CREDENTIALPROVIDER_VSTS_DEVICEFLOWTIMEOUTINSECONDS";
```
Comment (zarenner) [inline CredentialProvider.Microsoft/Util/EnvUtil.cs]: nit: For consistency with 'NUGET_CREDENTIALPROVIDER_VSTS_SESSIONTIMEMINUTES', drop the 'IN'

File: Build.props (hunk 1 above)
Comment (zarenner) [reply Build.props]: Should be

File: CredentialProvider.Microsoft/Resources.resx [hunk 4]
```diff
@@ -339,7 +339,11 @@ Cache Location
 Windows Integrated Authentication
     {13}
         Boolean to enable/disable using silent Windows Integrated Authentication
-        to authenticate as the logged-in user. Enabled by default.</value>
+        to authenticate as the logged-in user. Enabled by default.
+
+Device Flow Authentication Timeout
+    {14}
+        Device Flow authentication timeout in seconds. Default is 90 seconds.</value>
```
Comment (zarenner) [inline CredentialProvider.Microsoft/Resources.resx]: Please also update readme.md (and yes, it's annoying that it isn't automatically generated yet).

**Pull request merged**